### PR TITLE
feat: Options for Building and Running Private/Custom Models

### PR DESCRIPTION
**Pull request by ishaansehgal99**
**Reason for Change**:
This PR provides a README and example deployment for running a private/custom model. These are models not currently in [`supported_models.yaml`](https://github.com/Azure/kaito/blob/main/presets/models/supported_models.yaml)

PR also introduced new package `https://github.com/Azure/kaito/pkgs/container/kaito%2Fllm-reference-preset`

Resolves: https://github.com/Azure/kaito/issues/594

### CODE REVIEW

File: docs/custom-model-integration/Dockerfile.reference [hunk 1]
```diff
@@ -0,0 +1,39 @@
+FROM python:3.10-slim@sha256:684b1aaf96a7942b3c3af438d162e0baa3510aa7af25ad76d238e0c746bdec79
```
Comment (ishaansehgal99) [inline docs/custom-model-integration/Dockerfile.reference:1]: This file is how I built the reference image `ghcr.io/azure/kaito/llm-reference-preset:latest`. 

I can remove from PR, deemed not necessary

File: docs/custom-model-integration/custom-model-integration-guide.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+Navigate to the Kaito base directory and build the Docker image, including the weights directory in the build context:
+
+```sh
+docker build -t <IMAGE_NAME> --file docker/presets/models/tfs/Dockerfile --build-arg WEIGHTS_PATH=<WEIGHTS_PATH> --build-arg MODEL_TYPE=text-generation --build-arg VERSION=<VERSION> .
+
+docker push <IMAGE_NAME>
+```
+
+- Example Version: `0.0.1`
```
Comment (Fei-Guo) [inline docs/custom-model-integration/custom-model-integration-guide.md:46]: Move these before the `docker build`. Using the format:

For example,
export IMAGE_NAME="modelsregistry.azurecr.io/phi-3-mini-4k-instruct:0.0.1"
...

File: docs/custom-model-integration/Dockerfile.reference (hunk 1 above)
Comment (Fei-Guo) [reply docs/custom-model-integration/Dockerfile.reference:1]: I like it, we should keep it.

File: docs/custom-model-integration/Dockerfile.reference [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+# First, copy just the preset files and install dependencies
+# This is done before copying the code to utilize Docker's layer caching and
+# avoid reinstalling dependencies unless the requirements file changes.
+# Inference
+COPY presets/inference/${MODEL_TYPE}/requirements.txt /workspace/tfs/inference-requirements.txt
+RUN pip install --no-cache-dir -r inference-requirements.txt
+
+COPY presets/inference/${MODEL_TYPE}/inference_api.py /workspace/tfs/inference_api.py
+
+# Fine Tuning
```
Comment (Fei-Guo) [inline docs/custom-model-integration/Dockerfile.reference:25]: Do we need fine tuning part?

File: docs/custom-model-integration/Dockerfile.reference [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+COPY presets/tuning/${MODEL_TYPE}/cli.py /workspace/tfs/cli.py
+COPY presets/tuning/${MODEL_TYPE}/fine_tuning.py /workspace/tfs/fine_tuning.py
+COPY presets/tuning/${MODEL_TYPE}/parser.py /workspace/tfs/parser.py
+COPY presets/tuning/${MODEL_TYPE}/dataset.py /workspace/tfs/dataset.py
+
+# Copy the metrics server
+COPY presets/tuning/${MODEL_TYPE}/metrics/metrics_server.py /workspace/tfs/metrics_server.py
+
+# Remove Weights Copying
```
Comment (Fei-Guo) [inline docs/custom-model-integration/Dockerfile.reference:37]: Why do we need to copy weight file here?

File: docs/custom-model-integration/Dockerfile.reference (hunk 1 above)
Comment (Fei-Guo) [reply docs/custom-model-integration/Dockerfile.reference:1]: But please add the cmd line to build this image in the Makefile.

File: docs/custom-model-integration/Dockerfile.reference (hunk 4 above)
Comment (ishaansehgal99) [reply docs/custom-model-integration/Dockerfile.reference:37]: This was commented out

Comment (ishaansehgal99) [reply docs/custom-model-integration/Dockerfile.reference:37]: Removed

File: docs/custom-model-integration/Dockerfile.reference (hunk 3 above)
Comment (ishaansehgal99) [reply docs/custom-model-integration/Dockerfile.reference:25]: We do not....doesn't hurt to include
User would need to config file at /mnt/config and dataset at /mnt/data and tweak command line args slightly and they could run custom fine tuning

File: docs/custom-model-integration/Dockerfile.reference (hunk 1 above)
Comment (ishaansehgal99) [reply docs/custom-model-integration/Dockerfile.reference:1]: added

File: docs/custom-model-integration/custom-model-integration-guide.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+
+### Step 3: Build Docker Image with Private/Custom Weights
+
+Navigate to the Kaito base directory and build the Docker image, including the weights directory in the build context:
+
+```sh
+docker build -t <IMAGE_NAME> --file docker/presets/models/tfs/Dockerfile --build-arg WEIGHTS_PATH=<WEIGHTS_PATH> --build-arg MODEL_TYPE=text-generation --build-arg VERSION=<VERSION> .
+
+docker push <IMAGE_NAME>
```
[the remark this answers is not in the document]
Comment (ishaansehgal99) [reply docs/custom-model-integration/custom-model-integration-guide.md]: done

File: docs/custom-model-integration/Dockerfile.reference (hunk 3 above)
Comment (Fei-Guo) [reply docs/custom-model-integration/Dockerfile.reference:24]: I would say let us remove it for now. We'd add later if there is a need along with new docs about customized fine tuning.

**Review by Fei-Guo — APPROVED** [review]

**Pull request merged**